CMakeLists.txt

```diff
@@ -208,6 +210,35 @@ function(add_bundled_caf)
   add_subdirectory(caf EXCLUDE_FROM_ALL)
 endfunction()
 
+# Bundle prometheus-cpp if not building as subproject. Otherwise, the parent
+# project is responsible for adding the prometheus-cpp targets.
+if (NOT broker_is_subproject)
+  ZeekBundle_Add(
+    NAME prometheus-cpp
+    FETCH
+      GIT_REPOSITORY https://github.com/jupp0r/prometheus-cpp.git
+      GIT_TAG        v1.2.4
+    CONFIGURE
+      ENABLE_COMPRESSION=OFF
+      ENABLE_PUSH=OFF)
+elseif (NOT TARGET prometheus-cpp::core OR NOT TARGET prometheus-cpp::pull)
+  message(FATAL_ERROR "building Broker as sub-project requires prometheus-cpp targets")
+endif ()
+
+# Bundle all prometheus-cpp libraries that we need as single interface library.
+add_library(broker-prometheus-cpp INTERFACE)
+target_link_libraries(
+  broker-prometheus-cpp
+  INTERFACE
+    $<BUILD_INTERFACE:prometheus-cpp::core>
+    $<BUILD_INTERFACE:prometheus-cpp::pull>)
+# Prometheus-cpp sets C++ 11 via CMake property, but we need C++ 17.
+target_compile_features(broker-prometheus-cpp INTERFACE cxx_std_17)
+# Must be exported to keep CMake happy, even though we only need it internally.
+install(TARGETS broker-prometheus-cpp
+        EXPORT BrokerTargets
+        DESTINATION ${CMAKE_INSTALL_LIBDIR})
```

Comment:
Yes, simply dependending on the prometheus targets directly would be preferable. I've added this as a workaround because Broker won't compile otherwise.

The `prometheus-cpp` targets set the property `cxx_std_11`. On our end, we use [this script](https://github.com/zeek/cmake/blob/1be78cc8a889d95db047f473a0f48e0baee49f33/RequireCXX17.cmake) that will set `CMAKE_CXX_FLAGS` (I know...).

So what happens is that our targets will have the cxx flags applied to them. Then, CMake sees that the target "inherits" the `cxx_std_11` property and kindly adds something like `-std=c++11`. The end result is that CMake will pass `-std=c++17` and `-std=c++11` to the compiler. The compilers will use whatever comes last, so it tries to build Broker in C++11 mode, which will of course fail.

This extra target exists to pull in the prometheus targets but override the `cxx_std_11` property with `cxx_std_17` so that CMake will not select an old C++ version.

Instead of using this interface library, setting the `cxx_std_17` property on the three targets that pull in the prometheus library would probably work as well. I'm also thinking of kicking out the `RequireCXX17.cmake` script, but that should probably be a followup.